fighters/marth/src/status.rs

```diff
@@ -0,0 +1,156 @@
+use super::*;
+use globals::*;
+// status script import
+ 
+pub fn install() {
+    install_status_scripts!(
+        init_specials,
+    );
+}
+
+pub fn set_gravity_delay_resume_frame(energy: *mut FighterKineticEnergyGravity, frames: i32) {
+    unsafe {
+      *(energy as *mut i32).add(0x50 / 4) = frames;
+      *(energy as *mut bool).add(0x5C) = false;
+    }
+  }
+
+#[status_script(agent = "marth", status = FIGHTER_STATUS_KIND_SPECIAL_S, condition = LUA_SCRIPT_STATUS_FUNC_INIT_STATUS)]
+pub unsafe fn init_specials(fighter: &mut L2CFighterCommon, arg: u64) -> L2CValue {
+    let fighter_kind = WorkModule::get_int(fighter.module_accessor, *FIGHTER_INSTANCE_WORK_ID_INT_KIND);
+    let customize_special_hi_no = WorkModule::get_int(fighter.module_accessor, *FIGHTER_INSTANCE_WORK_ID_INT_CUSTOMIZE_SPECIAL_HI_NO);
+    let start_spd_x_mul = WorkModule::get_param_float(fighter.module_accessor, hash40("param_special_s"), hash40("start_spd_x_mul"));
+    let air_spd_y = WorkModule::get_param_float(fighter.module_accessor, hash40("param_special_s"), hash40("air_spd_y"));
+    let mut stop_energy = KineticModule::get_energy(fighter.module_accessor, *FIGHTER_KINETIC_ENERGY_ID_STOP) as *mut KineticEnergy;
+    let mut gravity_energy = KineticModule::get_energy(fighter.module_accessor, *FIGHTER_KINETIC_ENERGY_ID_GRAVITY) as *mut KineticEnergy;
+    let mut motion_energy = KineticModule::get_energy(fighter.module_accessor, *FIGHTER_KINETIC_ENERGY_ID_MOTION) as *mut KineticEnergy;
+
+    let mut aerial_y_speed = 0.0;
+    let mut aerial_x_speed = KineticModule::get_sum_speed_x(fighter.module_accessor, *KINETIC_ENERGY_RESERVE_ATTRIBUTE_MAIN) * start_spd_x_mul;
+
+    // [v] Disable motion energy
+    KineticModule::unable_energy(fighter.module_accessor, *FIGHTER_KINETIC_ENERGY_ID_MOTION);
+
+    let sum_speed_main = KineticModule::get_sum_speed(fighter.module_accessor, *KINETIC_ENERGY_RESERVE_ATTRIBUTE_MAIN);
+
+    // Check for the side B status you're currently in
+    let current_status_kind = StatusModule::status_kind(fighter.module_accessor);
+    let current_situation_kind = StatusModule::situation_kind(fighter.module_accessor);
+
+    // alStack192 = gravity energy
+    // alStack176 = stop energy
+    // alStack208 = motion energy
+    if current_status_kind == *FIGHTER_STATUS_KIND_SPECIAL_S {
```

Comment:
can't this big hairy couple of blocks be condensed down into a single block, more cleanly?

Reply:
Wanted to mirror the structure of the vanilla script from the decomp in case any issues arose and they could be more clearly identified in each particular block
Could probably condense the SPECIAL_S1/SPECIAL_S2 blocks and SPECIAL_S3/SPECIAL_S4 blocks together since they currently share the same logic